python/dgl/optim/pytorch/sparse_optim.py

```diff
@@ -119,9 +120,6 @@ def _comm_setup(self):
                 self._comm = nccl.Communicator(self._world_size,
                                                self._rank,
                                                nccl_id)
-                if self._rank == 0:
-                    # clear the store entry for future communicators
-                    store.delete_key('nccl_root_id')
```

Comment:
If we won't delete the key, I think we need to be something less generic than `nccl_root_id`. Maybe `SparseGradOptimizer.nccl_root_id`

Reply:
You mean, initialize it at the beginning? Make sense.

Reply:
`SparseGradOptimizer.nccl_root_id` sounds good